docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+print a different prompt each time the application starts up, before defaulting to a different prompt message
+for subsequent prompts.
+   * When the user starts the application, `isFirstPrompt` is set to `true` and thus, the prompt will be
+  ```
+  ----------------------------------------------------------------------
+  Now then, what can I do for you today?
+  (Need help? Type 'help' for a guide!)
+  ----------------------------------------------------------------------
+  >
+  ```
+  * Subsequent prompts in that app's session will be
+  ```
+  ----------------------------------------------------------------------
+  What's next?
+  ----------------------------------------------------------------------
+  >
+  ```
+
+---
+
+### Parsing User Input and Getting the Right Command
+**_TODO_**: Explain how the app parses user input and determines which `Command` subclass object to instantiate.
+
+---
+
+### Exercise
+#### List Exercise
+
+
+If the user's command type is to list the exercises available, i.e. `exercise /list`, the
+`Parser#parseUserInput(String userInput)` method will parse the 'exercise' base word and proceed to create exercise related
+command using `Parser#createExerciseCommand(String userInput)` method. This method will further evaluate the
+exercise action, in this case, `/list` and call the constructor of `ExerciseCommand` class by passing relevant parameters related to the
+ExerciseCommand constructor. If the exercise action is null or incorrect, an InvalidCommandException will be thrown. Once the exercise command is created,
+this exercise command is executed via the `ExerciseCommand#execute()` method. As it is executed, the method will check the
+type of action to be executed, in this case, list. It will then list the exercises available for selection from the exerciseList using the `ExerciseList#printExerciseList()`.
+
+The following sequence diagram illustrates how the `exercise /list` command works in greater detail:
+
+![List Exercise Sequence Diagram](uml/sequenceDiagrams/images/viewExercise.png)
+
+---
+
+### Workout
+#### Create New Workout
+
+A summary of the general procedure of a new workout being inputted and stored into WerkIt! is as follows:
+1. User enters the command `workout /new <workout name> /reps <number of repetitions>`.
+2. A new `Workout` object is created and stored in the application.
+3. The success response is printed to the user through the terminal.
+4. The new `Workout` object data is written to the resource file `workouts.txt`.
+
+The following sequence diagram illustrates how the `workout /new` command works in greater detail:
+> To simplify the sequence diagram, some method invocations that are deemed to be trivial 
+> have been removed from the sequence diagram. Some reference frames will be elaborated further 
+> down this section. Reference frames that will not be elaborated on will be made known.
+
+![Create Workout Sequence Diagram](uml/sequenceDiagrams/images/CreateWorkout.png)
```

Comment:
Same comment as previous sequence diagram on `parseUserInput()` invoking the constructor of `WorkoutCommand`.